File: free_tasks/ft_js/two_sums.js
class Solution {
    static twoSum(nums, target) {
        const prevDict = {}; // значення: індекс
        for (let i = 0; i < nums.length; i++) {
            const n = nums[i];
            const diff = target - n;
            if (prevDict[diff] !== undefined) {
                return [prevDict[diff], i];
            }
            prevDict[n] = i; // заповнення словника
        }
        return;
    }
}

const nums = [2, 7, 11, 15];
const target = 9;
console.log(Solution.twoSum(nums, target));

File: free_tasks/ft_py/two_sums.py
class Solution:
    nums = [2, 7, 11, 15]
    target = 9
    def two_Sum(nums, target):     
        prev_dict = {} # значення: індекс
        for i, n in enumerate(nums): # індекс та число в проіндексованому списку
            diff = target - n 
            if diff in prev_dict: 
                return [prev_dict[diff], i]
            prev_dict[n] = i # заповнення словника
        return
    print(two_Sum(nums, target))